File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public bool IsValidBST(TreeNode root) {
        bool valid = true;
        List<int> list = new();
        void Dfs(TreeNode root, ref bool validRef)
        {
            if (root is null)
            {
                return;
            }
            Dfs(root.left, ref validRef);
            if (list.Count > 0)
            {
                if (list[^1] >= root.val)
                    validRef = false;
            }
            list.Add(root.val);
            Dfs(root.right,ref validRef);
        }
        Dfs(root, ref valid);
        return valid;

    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        res = [0]
        def dfs(root,max_so_far):
            if not root:
                return
            
            if root.val >= max_so_far:
                res[0] += 1
                max_so_far = root.val
            
            dfs(root.left,max_so_far)
            dfs(root.right,max_so_far)
    
        dfs(root,-101)
        return res[0]



        